crates/matrix-sdk-ui/src/notification_client.rs

```diff
@@ -968,8 +968,12 @@ impl NotificationItem {
         if sender_display_name.is_none() || sender_avatar_url.is_none() {
             let sender_id = event.sender();
             for ev in state_events {
-                let Ok(ev) = ev.deserialize() else {
-                    continue;
+                let ev = match ev.deserialize() {
+                    Ok(ev) => ev,
+                    Err(error) => {
+                        warn!("Failed to deserialize state event: {:?}", error);
+                        continue;
+                    }
```

Comment:
I added this because figuring out what was wrong when generating the fake events was quite difficult. I can revert it, but I think it's useful.